src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+
+## Challenges and solutions
+
+Here are a few of the hurdles that needed to be overcome in order to achieve the
+team's stated goals. While some of these challenges are specific to Google
+Search, many of them are applicable to a wide range of sites that might be
+considering a service worker deployment.
+
+### Service worker overhead
+
+The biggest challenge, and the one true blocker for launching a service worker
+on Google Search, was to ensure that it did not do anything that might increase
+user-perceived latency. Google Search takes performance _very_ seriously, and in
+the past, has blocked launches of new functionality if it contributed even tens
+of milliseconds of additional latency for a given user population.
+
+When the team started collecting performance data during their earliest
+experiments, it became obvious that there would be a problem. The HTML returned
+in response to
+[navigation requests](https://developers.google.com/web/fundamentals/primers/service-workers/high-performance-loading#first_what_are_navigation_requests)
+for the search result page is dynamic, and varies greatly depending on logic
+that needs to run on Search's web servers. There's currently no way for the
+service worker to replicate this logic and return cached HTML immediately—the
+best it could do is to pass along navigation requests to the backend web
+servers, which necessitates a network request. Without a service worker, this
+network request would happen immediately upon user navigation, but when a
+service worker is registered, it will (by default) always need to be started up
+and given a chance to execute its [`fetch event handlers`](https://developers.google.com/web/fundamentals/primers/service-workers/#cache_and_return_requests),
+even when there's no chance that those `fetch` handlers will do
+anything other than go to the network. The amount of time that it takes to start
+up and run the service worker code is pure overhead added on top of every
+navigation, and puts the service worker implementation at too much of a latency
+disadvantage to justify any other benefits.
+
+![An illustration of the SW startup blocking the navigation request.](no-preload.png)
+
+Additionally, the team found that, based on measuring service worker boot times
+on real-world devices, there was a wide distribution of startup times, with some
+low-end mobile devices taking almost as much time to start up the service worker
+as it might take to make the network request for the results page's HTML.
+
+**Solution**
+
+The single, most crucial feature that allowed the Google Search team to move
+ahead with their service worker launch is
+[navigation preload](https://developers.google.com/web/updates/2017/02/navigation-preload).
+Using navigation preload is a key performance win for any service worker that
+needs to use a response from the network to satisfy navigation requests. It
+provides a hint to the browser to start making the navigation request
```

Comment:
Suggestion to introduce image with sentence, so like this:

It provides a hint to the browser to ... at the same time as the service worker starts up:

image here.

As long as the amount of time ...